app/src/main/java/com/gromov/diploma/view/analysis/AnalysisFragment.java

```diff
@@ -1,23 +1,316 @@
 package com.gromov.diploma.view.analysis;
 
+import android.app.DatePickerDialog;
+import android.app.TimePickerDialog;
+import android.content.Context;
+import android.graphics.Color;
 import android.os.Bundle;
+import android.support.design.widget.TextInputEditText;
 import android.support.v4.app.Fragment;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.text.format.DateUtils;
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.DatePicker;
+import android.widget.TimePicker;
 
+import com.github.mikephil.charting.charts.PieChart;
+import com.github.mikephil.charting.components.Legend;
+import com.github.mikephil.charting.data.PieData;
+import com.github.mikephil.charting.data.PieDataSet;
+import com.github.mikephil.charting.data.PieEntry;
+import com.github.mikephil.charting.formatter.PercentFormatter;
+import com.github.mikephil.charting.utils.ColorTemplate;
 import com.gromov.diploma.R;
+import com.gromov.diploma.data.async.AddMinDateAsyncTask;
+import com.gromov.diploma.data.async.GetCategoryAsyncTask;
+import com.gromov.diploma.data.async.GetPurchaseAsyncTask;
+import com.gromov.diploma.data.database.database.DatabasePurchase;
+import com.gromov.diploma.data.database.entities.Category;
+import com.gromov.diploma.data.database.entities.Product;
+import com.gromov.diploma.data.database.entities.Purchase;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
 
 
 public class AnalysisFragment extends Fragment {
 
+    private PieChart chart;
+    private DatabasePurchase databasePurchase;
+    private List<StatisticsByCategory> statistics;
+    private List<Category> categories;
+    private View v;
+    private Calendar calendarBegin = Calendar.getInstance();
+    private Calendar calendarEnd = Calendar.getInstance();
+    private Date dateBegin;
+    private Date dateEnd = Calendar.getInstance().getTime();
+    private Button dateBeginText;
+    private Button dateEndText;
+    private DatePicker d;
+    private RecyclerView recyclerView;
+    private AnalysisAdapter adapter;
```

Comment:
Field can be converted to a local variable more... (⌘F1)

Reply:
ok